context/app/static/js/hooks/useSelectItems.ts

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useReducer } from 'react';
+
+type SelectedItems = Set<string>;
+
+type SelectAction =
+  | {
+      type: 'addItem';
+      payload: string;
+    }
+  | {
+      type: 'removeItem';
+      payload: string;
+    }
+  | {
+      type: 'setSelectedItems';
+      payload: string[];
+    };
+
+function selectReducer(state: SelectedItems, action: SelectAction): SelectedItems {
+  switch (action.type) {
+    case 'addItem':
+      state.add(action.payload);
+      return new Set([...state]);
+    case 'removeItem':
+      state.delete(action.payload);
+      return new Set([...state]);
+    case 'setSelectedItems':
+      return new Set(action.payload);
+    default:
+      console.warn('Unrecognized action type', action);
+      return state;
+  }
+}
```

Comment:
Let's revisit this reducer - I think it's worth trying the `produce` approach with `toggleItem`, that should prevent the strict mode issue you experienced with the original implementation since it will keep the previous state Set untouched. This implementation masks the issues caused by using the `add` and `delete` functions on the previous state directly since calling them twice in a row since calling those operations with the same input on a Set twice doesn't have any additional effect, but it may still cause unexpected bugs later on since it mutates the previous state.

Reply:
Let's revise this reducer before merging

Reply:
Using a zustand store with initial state would require using `createStore` which seemed to be a lot of boilerplate for its worth and `immerReducer` also required creating a temp copy of draft to have a consistent return which somewhat defeats the purpose of introducing it.

Opted for the simplest approach to just add a temp copy of state to the existing reducer.